File: nmLabReport/lab27.c

#include <stdio.h>
#include <math.h>

// Function to integrate
double f(double x) {
    return exp(-x * x);
}

// Weights and abscissas for 2-point Gaussian quadrature
const double w[2] = {1.0, 1.0};
const double x[2] = {-0.5773502692, 0.5773502692};

double gaussianIntegration(double a, double b) {
    double integral = 0.0;
    double midpoint = (a + b) / 2.0;
    double half_length = (b - a) / 2.0;

    for (int i = 0; i < 2; i++) {
        integral += w[i] * f(midpoint + half_length * x[i]);
    }

    integral *= half_length;
    return integral;
}

int main() {
    double a = -1.0;
    double b = 1.0;
    double result = gaussianIntegration(a, b);
    printf("The integral from %f to %f is approximately %f\n", a, b, result);
    return 0;
}

File: nmLabReport/lab27.py
import math

def f(x):
    return math.exp(-x * x)

# Weights and abscissas for 2-point Gaussian quadrature
w = [1.0, 1.0]
x = [-0.5773502692, 0.5773502692]

def gaussian_integration(a, b):
    integral = 0.0
    midpoint = (a + b) / 2.0
    half_length = (b - a) / 2.0

    for i in range(2):
        integral += w[i] * f(midpoint + half_length * x[i])

    integral *= half_length
    return integral

def main():
    a = -1.0
    b = 1.0
    result = gaussian_integration(a, b)
    print(f"The integral from {a} to {b} is approximately {result}")

if __name__ == "__main__":
    main()